Treat method modifiers as required in Role::Tiny

s-nez opened the pull request:
Method modifiers like
```
around 'method' => sub {
    my $orig = shift;
    ucfirst $orig->(@_);
};
```
are technically required methods, since composing a role will fail if the modified method doesn't exist in the target class. This patch adds these to *requires* in Role::Tiny scanner.

tobyink commented:
This bit doesn't seem quite right:

```perl
map $_->[1], @{ $Role::Tiny::INFO{$role}{modifiers} or [] }
```

Have you considered this?

```
after qw( foo bar baz ) => sub { 1 };
```

s-nez commented:
You're right, I forgot about the ability to declare a modifier for multiple methods at once. I've updated the patch to handle this case as well as the arrayref version: ```before [ 'method1', 'method2' ] => sub { ... };```
I've also updated the test case to account for both of these.

I'm sorry for the omission.

tobyink commented:
Merged, but this is now an inconsistency with Moose and Mouse, so I'm hesitant to release it.

Also, it doesn't appear to consider this case:

```perl
package Local::Role;
use Role::Tiny;
after foo => sub { 1 };
sub foo { 1 };
```

The `foo` method shouldn't be listed in `requires`.

s-nez commented:
That is something that might require a general solution, since
```package Local::Role;
use Role::Tiny;
requires 'foo';
sub foo { 1 }
```
also exhibits this problem.

As for Moose/Mouse consistency, the behaviour is similar to Role::Tiny (composition dies if the modified method doesn't exist in target class), so it makes sense to extend them in a similar way.

tobyink commented:
I think this should all be covered by the latest release now.
